### ISSUE TITLE
After raspberry power off, cannot connect to accessories again.

### ISSUE THREAD

**Issue by himr3** [homebridge/HAP-NodeJS#201]
Hi, any solution to that problem, when power off or restart rpi, cannot connect again to the accessories until reset ios homekit settings or add new username to all accessory?

Thank You.

**Comment by husky-koglhof** [comment]
Hi, me too.
Following I've analyzed:

If i start from scratch, there will be min two connections 51826 (Default Port) and on additional 54826.
When i connect my iPad then 54826 will be used for pair-verify.
Now i can use the fully functionality of homekit.

If my iPad went to sleep or I disable WLAN for a while, homekit cannot reconnect to hap.
If i restart my hap, it only comes up with port 51826.

Within log files i did not see anything about communication from iPad to hap.

**Comment by KhaosT** [comment]
@husky-koglhof that's interesting, what's the router you're using? And when iPad comes back, can Bonjour Browser discover the hap service?

**Comment by husky-koglhof** [comment]
Hi, new interesting things.

On odroid (debian jessie) there's the problem, if i start the same on my mac mini (el capitan) there's no problem with it.

**Comment by KhaosT** [comment]
if that's the case, then it might be problem with avahi.

**Comment by husky-koglhof** [comment]
ok, so how i can fix this?

if you need my source, it's under https://github.com/husky-koglhof/ioBroker.homekit

**Comment by husky-koglhof** [comment]
ok, after restart of hap on my mac (and wifi off from iPad) now i have the same issue.
I did not see any communication on my network 

netstat -na |grep tcp|grep 5[14]8
tcp46      0      0  *.51826                *.*                    LISTEN     
tcp4       0      0  127.0.0.1.35480        *.*                    LISTEN

**Comment by KhaosT** [comment]
"wifi off from iPad"... what kind of communication are you expecting in this case ...?

**Comment by husky-koglhof** [comment]
I've disabled wireless lan on my ipad, cause this is the easiest way to test communication.

Normally i'm using my iPhone. But if i went out of wireless lan and come back, my iPhone can't reconnect to hap.

**Comment by KhaosT** [comment]
ok you still have your iPhone in the network. In that case, can you run [this](https://itunes.apple.com/us/app/inet-network-scanner/id340793353?mt=8) on your iPhone and see if iPhone can see the HomeKit service in Bonjour section?

**Comment by husky-koglhof** [comment]
Yes i can see Homekit (HAP) in Bonjour at iPhone.

I've tried to stop avahi on my odroid after shutdown of hap.
Now I can see each Accessory twice. One correct and one failed.

**Comment by KhaosT** [comment]
what do you mean by twice?

**Comment by husky-koglhof** [comment]
There are all Accessories shown double within the "eve Elgato" App.

I've now tested your BridgedCore.js.
I think I have something wrong within my Code, cause your Code is working correct on my odroid Machine.

I check this and give you feedback.

**Comment by husky-koglhof** [comment]
Hi, i think i've found the Problem, it's within my code.

Currently, following is implemented:

            sensor.addService(Service.HumiditySensor)
                .getCharacteristic(Characteristic.CurrentRelativeHumidity)
                .on('get', function (callback) {
                    var addr = p._id + "." + accessory['CurrentRelativeHumidity'][t];
                    adapter.log.debug('< hap ' + objName + ' get CurrentRelativeHumidity for ' + add);

                    adapter.getForeignState(addr, function (err, state) {
                        var value;
                        if (err || !state) {
                            value = 0;
                        } else {
                            value = parseFloat(state.val);
                        }
                        if (callback) callback(null, value);
                    });
                });

I think my function "adapter.getForeignState" needs to much time to come back and that's the Problem.

I've tested with hardcoded values and then there are no problems.

**Comment by husky-koglhof** [comment]
Hi, I've changed my Code today.

Now it's working correct. Reboot, wifi off, now it's all ok.

We can close this Issue from my Side.

Thanks a lot!

**Comment by flexzuu** [comment]
What exactly was your problem? I have a problem on reboot of my pi aswell. If i shutdown the process before rebooting everything works as intended. But if i just reboot with the node process running it crashes and on boot up next time it will start but my iPhone wont recognize it as the same bridge again. I have to delete the Bridge and readd it.

**Comment by husky-koglhof** [comment]
Hi, the Problem was that my subsystem needs too much time to callback the values.

I've analyzed it with Wireshark. With that tool, i've seen, that there are Requests from my iPad
to HAP but nothing comes back.
Then I've changed from subsystem to hardcoded values and it worked.

With the subsystem I've the same troubles as you. After a while, I can't reconnect to HAP.
Reboot doesn't work, restart of HAP also. 
The only way to get it to work was to remove the persistent Storefiles (AccessoryInfo*.json and IdentifierCache*.json), restart HAP and reset the homekit Settings at my iPad.

Now it's working correct.
